packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -63,22 +67,36 @@ async function handlePatientDiscoveryResults(
   pdResults: OutboundPatientDiscoveryResp[]
 ): Promise<void> {
   const { id, cxId } = patient;
-  const cqLinks = buildCQLinks(pdResults);
+  const cqLinks = await buildCQLinks(pdResults);
   if (cqLinks.length) await createOrUpdateCQPatientData({ id, cxId, cqLinks });
 }
 
-function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): CQLink[] {
-  return pdResults.flatMap(pd => {
-    const id = pd.externalGatewayPatient?.id;
-    const system = pd.externalGatewayPatient?.system;
-    const url = pd.gateway.url;
-    if (!id || !system || !url) return [];
-    return {
-      patientId: id,
-      systemId: system,
-      oid: pd.gateway.oid,
-      url,
-      id: pd.gateway.id,
-    };
-  });
+async function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): Promise<CQLink[]> {
+  const results = await Promise.all(
+    pdResults.map(async pd => {
+      const id = pd.externalGatewayPatient?.id;
+      const system = pd.externalGatewayPatient?.system;
+      const url = pd.gateway.url;
+      let patientExists = undefined;
+
+      if (isOutboundPDRespSuccessful(pd) && pd.patientResource && pd.cxId) {
+        const patientData = patientDataFromResource(pd.patientResource);
+        if (!patientData) return [];
+        patientExists = await getPatientByDemo({
+          cxId: pd.cxId,
+          demo: patientData,
+        });
+      }
+
+      if (!id || !system || !url || !patientExists) return [];
+      return {
+        patientId: id,
+        systemId: system,
+        oid: pd.gateway.oid,
+        url,
+        id: pd.gateway.id,
+      };
```

Comment:
We talked about storing the patient demographics returned w/ the link here, no? Or was it on the response table?